.github/workflows/dev_cd.yaml

```
@@ -1,49 +1,60 @@
 name: Dev CD
 
 on:
-  push:
-    branches:
-      - "main"
-      # you may add other branches to
-      # activate deployment for them, too.
-  workflow_dispatch: {}
+  release:
+    types: [published]
+      # trigger only on new release
 
 jobs:
-  get_commit_version:
+  verify_version:
     runs-on: ubuntu-latest
     outputs:
       # export to be used in other jobs
-      version: ${{ steps.get_commit_version.outputs.version }}
-      branch: ${{ steps.get_commit_version.outputs.branch }}
+      version: ${{ steps.get_version_tag.outputs.version }}
     steps:
       - uses: actions/checkout@v3
         name: Check out code
+
+      - uses: actions/setup-python@v4
+        name: Set up Python 3.x
         with:
-          # fetch the entire history
-          fetch-depth: 0
+          python-version: "3.x"
+
+      - id: get_version_tag
+        name: Get version tag
+        run: |
+          TAG_VER="${GITHUB_REF##*/}"
+          # set as output:
+          echo "version: ${TAG_VER}"
+          echo "version=${TAG_VER}" >> $GITHUB_OUTPUT
 
-      - id: get_commit_version
-        name: get commit version
+      - id: verify_semantic_tag_format
+        name: Verify tag format
+        # format must be compatible with semantic versioning
         run: |
-          BRANCH="$(git rev-parse --abbrev-ref HEAD)"
-          if git describe --tags
-          then
-            VERSION="$(git describe --tags)-${BRANCH}"
+          SEMVER_REGEX="^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(\-[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?(\+[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?$"
```

The regex suggested on https://semver.org/ is the following:

```suggestion
          SEMVER_REGEX="^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
```